systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java

```diff
@@ -200,14 +203,19 @@ public static JsonArray expectedServiceDiscoveryInfo(String plainAuth, String tl
 
     public static boolean checkLogForJSONFormat(Map<String, String> pods, String containerName) {
         boolean isJSON = false;
+        //this is only for decrease the number of records - kafka have record/line, operators record/11lines
+        String tail = "--tail=" + (containerName.contains("operator") ? "50" : "10");
 
         for (String podName : pods.keySet()) {
-            String logs = kubeClient().logs(podName, containerName).replaceFirst("([^{]+)", "");
+            String log = cmdKubeClient().execInCurrentNamespace(false, "logs", podName, "-c", containerName, tail).out();
+            Matcher matcher = JSON_PATTERN.matcher(log);
+            log = "[" + matcher.replaceAll("}, \\{").replaceAll("\n", " ").replaceFirst("(.*\\s)}, \\{", "{") + "]";
```

Comment:
This is not as efficient as it could be because the first `replaceAll()` returns a String, so subsequent `replaceAll/First()` will have to compile the regex on each invocation. Since you have this in a loop that potentially a lot of time wasted compiling the two patterns `N` times. You could use an explicit `Pattern.compile`, factored out into constants to avoid this overhead.

Also the `.replaceAll("\n", " ")` is probably unnecessary if you used `DOTALL` in the `replaceFirst()`.

Finally it's worth adding comments about what each replacement is doing to the output in order to make it valid JSON you can parse with `JsonArray()`.